examples/jax/jax-dist-spmd-mnist/spmd_mnist_classifier_fromscratch.py

```diff
@@ -0,0 +1,155 @@
+"""An MNIST example with single-program multiple-data (SPMD) data parallelism.
+
+The aim here is to illustrate how to use JAX's `pmap` to express and execute
+SPMD programs for data parallelism along a batch dimension, while also
+minimizing dependencies by avoiding the use of higher-level layers and
+optimizers libraries.
+"""
+
+import multiprocessing
+import os
+import time
+from functools import partial
+
+import numpy as np
+import numpy.random as npr
+
+# JAX will treat your CPU as a single device by default, regardless of the number
+# of cores available. Unfortunately, this means that using `pmap` is not possible out
+# of the box – we’ll first need to instruct JAX to split the CPU into multiple devices.
+# This variable has to be set before JAX or any library that imports it is imported
+
+os.environ["XLA_FLAGS"] = "--xla_force_host_platform_device_count={}".format(
+    multiprocessing.cpu_count()
+)
+
+import datasets  # noqa
+import jax  # noqa
+import jax.numpy as jnp  # noqa
+from jax import grad, jit, lax, pmap  # noqa
+from jax.scipy.special import logsumexp  # noqa
+from jax.tree_util import tree_map  # noqa
+
+jax.config.update("jax_cpu_collectives_implementation", "gloo")
+
+process_id = int(os.getenv("PROCESS_ID"))
+num_processes = int(os.getenv("NUM_PROCESSES"))
+coordinator_address = os.getenv("COORDINATOR_ADDRESS")
+coordinator_port = int(os.getenv("COORDINATOR_PORT"))
+coordinator_address = f"{coordinator_address}:{coordinator_port}"
```

Comment:
I think, you can directly populate the coordinator_address here, since you don't use port outside of the address.
```suggestion
coordinator_address = f"{os.getenv("COORDINATOR_ADDRESS")}:{int(os.getenv("COORDINATOR_PORT")}"
```